static/js/components/CustomizeOpenAIParameters.jsx

```diff
@@ -0,0 +1,237 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import EditOutlinedIcon from "@mui/icons-material/EditOutlined";
+import IconButton from "@mui/material/IconButton";
+import Tooltip from "@mui/material/Tooltip";
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import makeStyles from "@mui/styles/makeStyles";
+// eslint-disable-next-line import/no-unresolved
+import Form from "@rjsf/mui";
+import validator from "@rjsf/validator-ajv8";
+import DialogTitle from "@mui/material/DialogTitle";
+import * as profileActions from "../ducks/profile";
+
+const useStyles = makeStyles(() => ({
+  dialog: {
+    position: "fixed",
+  },
+  tooltip: {
+    fontSize: "1rem",
+    maxWidth: "30rem",
+  },
+}));
+
+const CustomizeOpenAIParameters = () => {
+  const dispatch = useDispatch();
+  const [aiopen, setAIOpen] = useState(false);
+  const classes = useStyles();
+
+  const site_openai_summary_parameters = useSelector(
+    (state) => state.config.openai_summary_parameters
+  );
+  const user_openai_summary_parameters = useSelector(
+    (state) => state.profile.preferences.summary.OpenAI
+  );
+
+  const default_openai_summary_parameters = {
+    ...site_openai_summary_parameters,
+    ...user_openai_summary_parameters,
+  };
+
+  const handleAIClickOpen = () => {
+    setAIOpen(true);
+  };
+
+  const handleAIClose = () => {
+    setAIOpen(false);
+  };
+
+  const handleAISubmit = ({ formData }) => {
+    const prefs = {
+      summary: {
+        OpenAI: {
+          active: user_openai_summary_parameters.active,
+          ...formData,
+        },
+      },
+    };
+    dispatch(profileActions.updateUserPreferences(prefs));
+    handleAIClose();
+  };
+
+  const formSchema = {
+    type: "object",
+    properties: {
+      model: {
+        type: "string",
+        title: "model",
+        examples: [
+          "gpt-4",
+          "gpt-3.5-turbo",
+          "text-davinci-003",
+          "davinci",
+          "gpt-4-32k",
+        ],
+      },
+      prompt: {
+        type: "string",
+        title: "prompt",
+      },
+      temperature: {
+        type: "number",
+        title: "Temperature",
+        minimum: 0.0,
+        maximum: 1.0,
+        multipleOf: 0.05,
+      },
+      max_tokens: {
+        type: "integer",
+        title: "max_tokens",
+        minimum: 10,
+        maximum: 1000,
```

Comment:
The models allow for longer outputs than 1000 tokens but we're trying to create succinct summaries.